hackathon_site/event/api_views.py

```diff
@@ -166,16 +167,33 @@ def patch(self, request, *args, **kwargs):
 class CurrentProfileView(mixins.UpdateModelMixin, generics.GenericAPIView):
 
     queryset = Profile.objects.all()
-    serializer_class = CurrentProfileSerializer
-    permission_classes = [UserHasProfile]
+
+    def get_serializer_class(self):
+        if self.request.method == "PATCH":
+            return CurrentProfileSerializer
+        elif self.request.method == "POST":
+            return CreateProfileSerializer
 
     def get_object(self):
         return self.request.user.profile
+    def get_permissions(self):
+        if self.request.method == "PATCH":
+            return [UserHasProfile()]
+        elif self.request.method == "POST":
```

Comment:
you need to add GET to the if statement.
and the permissions for the post method is whether the user is authenticated, you can use `[permissions.IsAuthenticated()]` here